### PR TITLE
Combined default and non-default share creation into one template function

### PR DESCRIPTION
**Pull request by imran-ansari**
create_share_on_fpg() can now create both default and non-default shares.
It achieves this by using specific implementations of two functions the references of which are passed to it as argument. Rest of the function steps are common to both default and non-default share creation process.

### CODE REVIEW

**Comment by imran-ansari** [comment]
@wdurairaj / @nilangekarss - please review the changes. I would like the testers to start using this one ASAP as this has rollback for non-default share creation process including some fixes.

File: hpedockerplugin/file_manager.py [hunk 1]
```diff
@@ -179,6 +129,47 @@ def create_share(self, share_name, **args):
...
+        try:
+            fpg_info = self._fp_etcd_client.get_fpg_metadata(
+                self._backend,
+                cpg_name, fpg_name
+            )
+        except exception.EtcdMetadataNotFound:
+            LOG.info("Specified FPG %s not found in ETCD. Checking "
+                     "if this is a legacy FPG..." % fpg_name)
+            # Assume it's a legacy FPG, try to get details
+            leg_fpg = self._hpeplugin_driver.get_fpg(fpg_name)
```
Comment (nilangekarss) [inline hpedockerplugin/file_manager.py]: Hi Imran, what if the fpg doesn't exist on 3par itself?

Comment (imran-ansari) [reply hpedockerplugin/file_manager.py]: Then the plugin assumes that user wants to create a new FPG by the specified name

**Comment by nilangekarss** [comment]
Merging this code so that testers can pick this up for their testing. We will document the use cases with their commands and expected execution result. Any bugs will be taken as new pull request.

**Pull request merged**